pkg/operator/controller/config_controller.go

```diff
@@ -161,3 +170,68 @@ func enqueueWatched(_ context.Context, _ client.Object) []ctrl.Request {
 		},
 	}
 }
+
+// managePrometheusMonitoring manages the lifecycle of the Prometheus monitoring goroutine
+func (r *ConfigReconciler) managePrometheusMonitoring(ctx context.Context, kaiConfig *kaiv1.Config) {
+	// Check if external Prometheus is configured
+	hasExternalPrometheus := kaiConfig.Spec.Prometheus != nil &&
+		kaiConfig.Spec.Prometheus.ExternalPrometheusUrl != nil &&
+		*kaiConfig.Spec.Prometheus.ExternalPrometheusUrl != ""
+
+	if hasExternalPrometheus {
```

Comment:
In the context of this function, the happy flow is "hasExternalPrometheus". So just invert this condition, something like:
```Go
if !hasExternalPrometheus {
	if r.monitoringCancel != nil {
			r.monitoringCancel()
			r.monitoringCtx = nil
			r.monitoringCancel = nil
		}
	return
}

This will keep the rest of the code less indented, and make the function more readable.
In general: keep happy flow to the left (un-nested) as much as possible